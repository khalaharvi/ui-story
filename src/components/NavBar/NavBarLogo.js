import React from 'react';
import PropTypes from 'prop-types';

import { createClassNames } from 'utils';
import Link from 'components/Link';
import Image from 'components/Image';

const cx = createClassNames('navbar-logo');

// TO DO: allow for custom width/height

const NavBarLogo = ({
  alt,
  logo,
  src,
  to
}) => {
  const imgOrElement = typeof logo !== 'undefined' ? logo : <Image src={src} alt={alt} />;

  return (
    <div className={cx('')}>
      {to ? <Link to={to}>{imgOrElement}</Link> : imgOrElement}
    </div>
  );
};

NavBarLogo.propTypes = {
  alt: (props, propName) => {
    const alt = props[propName];

    if (typeof props.logo === 'undefined' && typeof alt !== 'string') {
      throw new Error("'alt' is required.");
    }
  },
  logo: (props, propName) => {
    const logo = props[propName];

    if (typeof props.src === 'undefined' && !React.isValidElement(logo)) {
      throw new Error("'logo' is required and must be a valid React element.");
    }
  },
  src: (props, propName) => {
    const src = props[propName];

    if (typeof props.logo === 'undefined' && typeof src !== 'string') {
      throw new Error("'src' is required.");
    }
  },
  to: PropTypes.string
};

export default NavBarLogo;
