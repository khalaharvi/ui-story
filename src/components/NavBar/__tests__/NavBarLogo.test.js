import React from 'react';
import { mount, shallow } from 'enzyme';

import NavBarLogo from '../NavBarLogo';

describe('<NavBarLogo />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<NavBarLogo src="https://placehold.it/125x35" alt="Logo" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with a custom element', () => {
    const Logo = () => (<img src="https://placehold.it/125x35" alt="Logo" />);
    const wrapper = shallow(<NavBarLogo logo={<Logo />} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an <a> when to prop is passed', () => {
    const wrapper = mount(<NavBarLogo src="https://placehold.it/125x35" alt="Logo" to="https://deringhall.com" />);

    const a = wrapper.find('a');

    expect(a).toHaveLength(1);

    wrapper.unmount();
  });

  it('should have an <a> with a custom element when to prop is passed', () => {
    const Logo = () => (<img src="https://placehold.it/125x35" alt="Logo" />);
    const wrapper = mount(<NavBarLogo logo={<Logo />} to="https://deringhall.com" />);

    const a = wrapper.find('a');

    expect(a).toHaveLength(1);

    wrapper.unmount();
  });
});
