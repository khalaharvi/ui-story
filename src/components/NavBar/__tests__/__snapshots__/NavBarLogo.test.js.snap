// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBarLogo /> renders correctly 1`] = `
<div
  className="dh-navbar-logo"
>
  <Image
    alt="Logo"
    src="https://placehold.it/125x35"
  />
</div>
`;

exports[`<NavBarLogo /> renders correctly with a custom element 1`] = `
<div
  className="dh-navbar-logo"
>
  <Logo />
</div>
`;
